### ISSUE TITLE
Output raw data

### ISSUE THREAD

**Issue by biorelated** [wdecoster/NanoPlot#11]
Thank you for a wonderful tool! Could there be an option to output raw data used to generate these plots. This way,  users can create/ combine multiple plots using  other statistical tools and explore the data a little more.  I would like to see a summary file that contains: 

1. readname, median_basequality, average_basequality, read_length

maybe you could toggle a verbose argument that allows the tool to spit these data out.

**Comment by wdecoster** [comment]
Hi biorelated,

Thanks for the suggestion. Outputting raw data wouldn't be too hard to add, I'll have a look at it soon.  
However, my script doesn't keep the readnames and doesn't calculate the median basequality, so this may not exactly be what you are looking for.

I guess I could write a new script which does just that - extraction of `readname, median_basequality, average_basequality, read_length`. Are you using fastq files as input?

> This way, users can create/ combine multiple plots using other statistical tools and explore the data a little more.

If you have suggestions which plots to add I would be more than happy to take a look at those too!

Cheers,  
Wouter

**Comment by biorelated** [comment]
Thank you for your prompt reply. 

Yes I am using fastq files as input. In addition, I have a barcoded set or runs which I would like to summarise . Average quality per read is not very informative given that some regions might have consistently low or high qualities in some regions, thus it would be nice to also have a look at the median values. 

I guess extracting the read-name would be useful for summarising the data per read. It allows someone to filter the data or combine it with other metadata in a more flexible way.  For example I want to plot a correlation between read-length or median basequality scores and viral copies or ct-values, from a separate metadata file.

 I tend to prefer  the idea of decoupling data generation from summary plots.  I can see how I can use nanoplot to generate the data and optionally use another tool for summarising the data

**Comment by wdecoster** [comment]
Indeed, having low or high qualities in some regions might skew the average making the median more robust. However, I would argue that since extreme outliers are not possible (scale is roughly between 3 and  38) the average is a reasonable estimator. But fair enough, it makes sense that you would like to work with the medians.

Given that NanoPlot currently does not extract names and medians it makes more sense for me to put your request in a separate script.
To summarize: you would like to feed in a fastq and receive as output a tab separated file containing readname, median_basequality, average_basequality, read_length.

I think I will have time to write that later this week. It's not too hard.

**Comment by biorelated** [comment]
Many thanks!

**Comment by wdecoster** [comment]
Hi biorelated,

I just wrote fastqmetrics (https://github.com/wdecoster/fastqmetrics). Can you see if that works for you?
Hasn't been tested extensively but works on my system.

If you have further feedback/suggestions/issues or feature requests please open an issue on the GitHub of fastqmetrics.

Cheers,
Wouter

**Comment by wdecoster** [comment]
As a follow-up on this: NanoPlot can since recently store the extracted metrics in a pickle file using the `--store` flag. The (binary) pickle file contains the pandas DataFrame containing all extracted metrics.
 You can rerun the plotting with the same data using the `--pickle` input option, or use the data in the pickle for your own plots.
